### PR TITLE
Add jsonformoderncpp

### PR DESCRIPTION
**Pull request by IceflowRE**
Specify library name and version:  **jsonformoderncpp/3.7.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

My first contribution here, so please review carefully, it will help me for the next packages.

---

Regarding `No Beta user` https://github.com/conan-io/conan-center-index/issues/4#issuecomment-535867892

### CODE REVIEW

File: recipes/jsonformoderncpp/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    author = "Bincrafters <bincrafters@gmail.com>"
+    license = "MIT"
+
+    options = {
+        "multiple_headers": [True, False],
+    }
+    default_options = {
+        "multiple_headers": False,
+    }
+    exports = ["LICENSE.md"]
```
Comment (Croydon) [inline recipes/jsonformoderncpp/all/conanfile.py]: please remove this

File: recipes/jsonformoderncpp/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
```
Comment (Croydon) [inline recipes/jsonformoderncpp/all/conanfile.py]: please remove this two meta lines

**Comment by Croydon** [comment]
The no EAP error might come once again from upper/lowercase mismatch?

**Comment by uilianries** [comment]
@Croydon I'mm curious about it, because @IceflowRE has been accepted 1 week ago: 

https://github.com/conan-io/conan-center-index/issues/4#issuecomment-535867892

@lasote Do you know something about this EAP error?

**Comment by lasote** [comment]
There was a typo in the whitelisted users. Sorry, Now you are in.

File: recipes/jsonformoderncpp/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    name = "jsonformoderncpp"
+    homepage = "https://github.com/nlohmann/json"
+    description = "JSON for Modern C++ parser and generator."
+    topics = ("conan", "jsonformoderncpp", "json", "header-only")
+    url = "https://github.com/conan-io/conan-center-index"
+    author = "Bincrafters <bincrafters@gmail.com>"
+    license = "MIT"
+
+    options = {
+        "multiple_headers": [True, False],
```
Comment (lasote) [inline recipes/jsonformoderncpp/all/conanfile.py]: You are missing a default value for the option. [We encourage to test the recipes locally first](https://github.com/conan-io/conan-center-index/wiki#test-the-recipe-locally).

What is the `JSON_MultipleHeaders` about?

Comment (IceflowRE) [reply recipes/jsonformoderncpp/all/conanfile.py]: The local test was successful at the time tested.
I will resolve later these days.

Comment (IceflowRE) [reply recipes/jsonformoderncpp/all/conanfile.py]: Multiple headers are defining if one header file or multiple are used afaik.

Comment (Croydon) [reply recipes/jsonformoderncpp/all/conanfile.py]: Sorry, seems like my comment was confusing when combined with how GitHub is displaying inline comments. 

I meant to only remove the `exports = ["LICENSE.md"]` line :)

**Comment by danimtb** [comment]
I have added some changes to conform to the new checks we recently introduced in the Conan Center hook. Hopefully, this will fix the remaining issues and we can get this recipe included. Thanks a lot @IceflowRE!

**Comment by danimtb** [comment]
I think there might be a flag missing to set the `cppstd` as the test_package fails to build

**Comment by lasote** [comment]
A similar case happened with catch2: https://github.com/conan-io/conan-center-index/pull/51/files
In that case, until we find a better or more solid fix, we enabled the CXX standard in the `test_package/CMakeLists.txt` file:

```
set(CMAKE_CXX_STANDARD 11)
set(CMAKE_CXX_STANDARD_REQUIRED ON)
```

We are going to include a new helper in Conan 1.20 to compare better the possibles values of the `compiler.cppstd` and being able to raise InvalidConfiguration, but still not there.

**Comment by IceflowRE** [comment]
Squashed and signed the commits.

File: recipes/jsonformoderncpp/all/test_package/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,10 @@
+cmake_minimum_required(VERSION 2.8.12)
+project(test_package)
+
+set(CMAKE_VERBOSE_MAKEFILE TRUE)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
```
Comment (uilianries) [inline recipes/jsonformoderncpp/all/test_package/CMakeLists.txt]:
```suggestion
target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 11)
```

**Pull request merged**